app/services/finance/npq/assurance_report/query.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Finance
+  module NPQ
+    class AssuranceReport
+      class Query
+        def initialize(statement)
+          self.statement = statement
+        end
+
+        def participant_declarations
+          @participant_declarations ||= ParticipantDeclaration::NPQ.find_by_sql(sql)
+        end
+
+      private
+
+        attr_accessor :statement
+
+        def sql
+          <<~EOSQL
```

Comment:
same here, can we add a test for this service?

Reply:
Fair point, yes the `CASE WHEN` could certainly do with a spec.